frontend/src/pages/site/components/header/header.tsx

```diff
@@ -1,19 +1,98 @@
-import { type SiteHeaderContent } from '~/packages/sections/sections.js';
+import { Input } from '~/libs/components/components.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useAppForm, useCallback, useState } from '~/libs/hooks/hooks.js';
+import { siteHeaderUpdateContentValidationSchema } from '~/packages/sections/sections.js';
+import {
+  type SiteHeaderContent,
+  type SiteHeaderUpdateContentDto,
+} from '~/packages/sections/sections.js';
+import { Overlay } from '~/pages/site/components/overlay/overlay.js';
 
 import styles from './styles.module.scss';
 
 type Properties = {
   content: SiteHeaderContent;
+  onUpdate: (payload: { logo: string; phone: string }) => void;
 };
 
 const Header: React.FC<Properties> = ({
   content: { logo, phone },
+  onUpdate,
 }: Properties) => {
+  const { control, errors, handleSubmit, handleReset } =
+    useAppForm<SiteHeaderUpdateContentDto>({
+      defaultValues: { logo, phone },
+      validationSchema: siteHeaderUpdateContentValidationSchema,
+    });
+
+  const [showOverlay, setShowOverlay] = useState<boolean>(false);
```

Comment:
let's create a wrapper component and/or hook and move all logic there. since we will need reuse it for all sections